src/storage/storage.ts

```diff
@@ -1023,4 +1023,13 @@ export class Storage implements IStorage {
   async unregisterNanoContract(ncId: string): Promise<void> {
     return this.store.unregisterNanoContract(ncId);
   }
+
+  /**
+   * Update nano contract registered address
+   * @param ncId Nano Contract ID.
+   * @param address New registered address
+   */
+  async updateNanoContractRegisteredAddress(ncId: string, address: string): Promise<void> {
+    return this.store.updateNanoContractRegisteredAddress(ncId, address);
+  }
```

Comment:
The registered address may not be a wallet address.
Do you think we should use the address index instead? This would ensure the address was ours.

We can also check here with `this.isAddressMine(address)` so we don't add an address from outside the wallet (or an invalid one).

Reply:
suggestion: Even if the address should not necessarily be from the current wallet, an assertion that the parameter is at least a _valid Base58 address_ is important.

Reply:
This validation is currently done in the wallet using this method, but I guess we can add a protection here as well

Reply:
Done https://github.com/HathorNetwork/hathor-wallet-lib/pull/657/commits/6064a6882a946cc3aa8cc86d54fb90f3032aa667